### PR TITLE
Only use the RFC6962 hasher in Migrillian.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: trillian/migrillian/core/trillian.go [hunk 1]
```diff
@@ -61,12 +60,7 @@ func NewPreorderedLogClient(
 	if got, want := tree.TreeType, trillian.TreeType_PREORDERED_LOG; got != want {
 		return nil, fmt.Errorf("tree %d is %v, want %v", tree.TreeId, got, want)
 	}
-	// If this isn't an RFC6962 hasher, Questions Should Be Asked.
-	logHasher, err := registry.NewLogHasher(tree.HashStrategy)
```
Comment (pavelkalinnikov) [inline trillian/migrillian/core/trillian.go:65]: Maybe add a check here that `tree.HashStrategy == RFC6962`?

Comment (pphaneuf) [reply trillian/migrillian/core/trillian.go:65]: Done.

File: trillian/migrillian/core/trillian.go [hunk 2]
```diff
@@ -42,8 +41,8 @@ var errRetry = errors.New("retry")
 // pre-ordered log tree.
 type PreorderedLogClient struct {
 	cli    trillian.TrillianLogClient
-	hasher hashers.LogHasher
-	tree   *trillian.Tree
+	hasher *hasher.Hasher
+	logID  int64
```
Comment (pavelkalinnikov) [inline trillian/migrillian/core/trillian.go:45]: nit: I'd prefer `treeID` for consistency.

**Review by pavelkalinnikov — APPROVED** [review]
LGTM % optional nit.

File: trillian/migrillian/core/trillian.go (hunk 2 above)
Comment (pphaneuf) [reply trillian/migrillian/core/trillian.go]: It said TreeId where it's set, and LogId where it's used, had to pick the wrong one!

Comment (pavelkalinnikov) [reply trillian/migrillian/core/trillian.go]: Indeed. I guess both options are equally inconsistent :)

**Pull request merged**